src/Cache/Engine/FileEngine.php

```diff
@@ -314,24 +316,22 @@ protected function _clearDirectory($path, $now, $threshold)
 
             if ($threshold) {
                 $mtime = $file->getMTime();
-
                 if ($mtime > $threshold) {
                     continue;
                 }
-                $expires = (int)$file->current();
 
+                $expires = (int)$file->current();
                 if ($expires > $now) {
                     continue;
                 }
             }
-            if ($file->isFile()) {
```

Comment:
if `SplFileObject($path . $entry, 'r')` went wrong, it `continue`
No matter `$path . $entry` is not exists or unreadable.